src/pybind/mgr/cephadm/schedule.py

```diff
@@ -42,8 +42,10 @@ def place(self, host_pool, count=None):
         if not host_pool:
             return []
         host_pool = [x for x in host_pool]
+        # gen seed off of self.spec to make shuffling deterministic
+        seed = hash(self.spec)
         # shuffle for pseudo random selection
-        random.shuffle(host_pool)
+        random.Random(seed).shuffle(host_pool)
```

Comment:
Instead of making the scheduler deterministic for all eternity without an escape route, what about instead return something to the user that is independent of the actual placement? Maybe `"Will place {count} daemons on any of those hosts: {hosts}"`?

Reply:
If you have 100 potential candidates this wouldn't be very meaningful though..

Do you see any disadvantages for the scheduler to be deterministic?

Reply:
Had a small chat offline and agreed to make the `SimpleScheduler` deterministic as we couldn't find an actual downside.